syncer_test.go

```diff
@@ -152,6 +155,63 @@ func TestSyncerRunSuccess(t *testing.T) {
 	}
 }
 
+func TestSyncChangedSecrets(t *testing.T) {
+	type secret struct {
+		Name   string `json:"name"`
+		Secret string `json:"secret"`
+	}
+
+	// Create a new server that generates a random value for a secret named changing-secret every time
+	// a hander is called. This lets us simulate secret changes.
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Helper()
+
+		bytes := make([]byte, 8)
+		if _, err := rand.Read(bytes); err != nil {
+			t.Fatalf("rand.Read failed: %v", err)
+		}
+		s := &secret{
+			Name:   "changing-secret",
+			Secret: hex.EncodeToString(bytes),
+		}
+
+		encoder := json.NewEncoder(w)
+		var err error
+		switch r.URL.Path {
+		case "/secrets":
+			s.Secret = ""
+			err = encoder.Encode([]*secret{s})
+		case "/secret/changing-secret":
```

Comment:
nit: might be good to have a variable named `changing-secret` so that it's clear that this path is not hardcoded, but rather that it's `/secret/<special variable>`

Reply:
I created the handler path with `"/secret/" + s.Name`. I think that makes it more clear.